### ISSUE TITLE
[2.1.0] no such file or directory, lstat '/var/folders/xxxxx/T/carbon-multi-page@2.1.0/build'

### ISSUE THREAD

**Issue by efstathiosntonas** [magidoc-org/magidoc#98]
After upgrading to 2.1.0 it throws this error:

```bash
✔ Determine tmp directories [0.0s]
✔ Select Package Manager [0.1s]
  › Selected pnpm
↓ Template already unzipped. [SKIPPED]
↓ Template already unzipped [SKIPPED]
✔ Install dependencies [0.6s]
✔ Resolving template configuration [0.0s]
  › Found 7 supported keys
    Target schema location: ./src/_schema.json
✔ Load GraphQL Schema [0.7s]
✔ Write variables file [0.0s]
✔ Build template [1s]
✖ Move output

------- Stacktrace -------
[Error: ENOENT: no such file or directory, lstat '/var/folders/yz/dk5dbx8n2hqcfk1kdktxmknh0000gn/T/carbon-multi-page@2.1.0/build'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'lstat',
  path: '/var/folders/yz/dk5dbx8n2hqcfk1kdktxmknh0000gn/T/carbon-multi-page@2.1.0/build'
}
```

I'm using node 14.9, latest pnpm and lib version 2.1.0. 

It seems that `Build Template` step took 1s to finish while on `v2.0.0` it took like a minute to finish.

edit: I totally removed the `website.options.queryGenerationFactories`, runned `magidoc generate` and it didn't throw an error about missing scalars, it went directly to the error described above since the scalars check is happening during `Build template` step.

edit 2: I've manually created the `build` folder, runned `magidoc generate`, (the process was unexpectedly really fast), runned `magidoc preview`, visited `localhost:4000` and the page was blank (as expected since the Build template step took only 1 sec to finish).

edit 3: don't know if it matters but my global packages are installed in $`HOME/.npm-global` folder instead of npm default one so I won't have to use `sudo` while installing global packages.

**Comment by pelletier197** [comment]
I've updated some dependencies with the update, and I've had an issue like this one before. It may be related to a ton of things.

- bug in lib update
- use npm, while magidoc is tested with pnpm
- node version 14, while magidoc is tested on 16
- different os 
- etc

Since you tested yesterday and it worked, my guess is a lib update. Let me see if I can reproduce, maybe I'll have to revert a dependency like last time.

> It's really an annoying issue, because I have a post deployment step that generates both the documentation and the demo website with the latest magidoc release, and both these steps succeeded

**Comment by pelletier197** [comment]
@efstathiosntonas Managed to reproduce when using node 14. I'll see what's causing the issue for node 14. If you upgrade to node 16, it should fix the issue. 

```
npm 6.14.17
node v14.19.3
```

I'm questioning if it would be better to make 16 (the LTS) the minimal supported version, or if it's worth supporting earlier versions of node as well. Got any opinion on that?

**Comment by efstathiosntonas** [comment]
I cannot use node 16 due to bad firebase support on monorepos and workspaces config. I could use nvm but this would require lot of changes on my environment in general. It’s up to you if you decide to support v16+ 😁

**Comment by pelletier197** [comment]
I actually tried using node 14 to develop locally, and `svelte-kit`, the engine used to create the website, forces node `16.7+`. (See [this line](https://github.com/sveltejs/kit/blob/4e4625ea6d9a084bc767ae216704aacd95fe8730/packages/kit/package.json#L95)). 

So unfortunately, I think it's not even up to me, I don't think i'll be able to make it work under node 14. Although, i'm really surprised it worked yesterday to be honest, because version 2.0.0 with node 14 gives me the same error with `2.0.0`.

Anyway, I hope you find a way to integrate it in your project eventually. We use it where I work, but It's still an early project, and I want to gain more users like you to improve it even further. Thanks a lot for your contribution! 

----

For the record, the issue seems to be un unhandled error in svelte-kit when using node 14 
```
node:425300) UnhandledPromiseRejectionWarning: file:///tmp/carbon-multi-page@2.1.0/node_modules/@sveltejs/kit/dist/node/polyfills.js:11
import { webcrypto } from 'crypto';
         ^^^^^^^^^
SyntaxError: The requested module 'crypto' does not provide an export named 'webcrypto'
```

**Comment by efstathiosntonas** [comment]
It worked yesterday because I had node 16 installed, then I faced the firebase issue (witch is known for 2 months by the firebase team and still unresolved) and today I’ve switched back to 14. 

I’ll see what I can do, I want to ditch v14 for many reasons, probably I’ll switch to nvm.

**Comment by efstathiosntonas** [comment]
this is the [issue](https://github.com/firebase/firebase-functions/issues/1050) with firebase 16 and monorepos, linking it here for visibility.

**Comment by pelletier197** [comment]
I will release a new version with the `engine.node` set, normally npm would give you an error if your try to install using an invalid version. 

I'll add it to the docs as well. Thanks!

**Comment by akkikumar72** [comment]
Getting this same error when trying to generate in vercel dashboard.

Using:
- @magidoc/cli 2.12.0
- pnpm 7

![Screenshot 2022-09-10 at 09 25 44](https://user-images.githubusercontent.com/4746256/189473711-c479ff57-bb6e-4094-b52b-19c5151dcd3d.png)

**Comment by pelletier197** [comment]
Thanks for letting me know, I will have a look

**Comment by pelletier197** [comment]
@akkikumar72 what node version are you on?

**Comment by akkikumar72** [comment]
@pelletier197 - according to vercel they support v16.14. https://vercel.com/changelog/node-js-16-lts-is-now-available
So its using node v16.14

**Comment by pelletier197** [comment]
Okay. Then I'm a little confused. I was not able to reproduce the error myself. I will dig a little more later and see if I can reproduce.

**Comment by efstathiosntonas** [comment]
@akkikumar72 can you run it with `—stacktrace` (as stated on the screenshot) and paste the output?

**Comment by akkikumar72** [comment]
For some reason. --stacktrace doesn't work in custom `vercel build`

**Comment by akkikumar72** [comment]
Let me try to give a reproducible repo here. I tried to run multi-schema example in vercel.

**Comment by akkikumar72** [comment]
https://github.com/akkikumar72/docs-magicDoc try to deploy this in vercel.

vercel - https://vercel.com/docs/cli

**Comment by pelletier197** [comment]
Thanks, I will have a look as soon as possible.

**Comment by pelletier197** [comment]
Thank you a lot for the repro project. I managed to reproduce the issue and figure out what was the problem. Apparently, Svelte-kit seem to do something special when it detects it's deployed on vercel, and it automatically outputs build files somewhere else than what is specified in the configuration....  very weird and not very well documented feature.

I already opened and merged a bugfix for that (2.13.1), so thank you. 

For the record, I managed to deploy your website [here](https://magidoc-pelletier197.vercel.app/) with a few tweaks
- I needed to add a `vercel.json` file with the following content for URLs to be resolved.
```json
{
    "cleanUrls": true
}
```
I will make sure to document that as part of the `deployment` documentation.

- There was a typo in your `index.html` (anolist rather than anilist).
